apps/flame-defi/app/features/evm-wallet/services/non-fungible-position-manager-service/index.ts

```diff
@@ -350,6 +415,343 @@ export class NonfungiblePositionManagerService extends GenericContractService {
 
     return positions;
   }
+
+  /**
+   * Extended write contract method that allows specifying gas limit
+   * @private
+   */
+  private async writeContractWithGasLimit(
+    chainId: number,
+    methodName: string,
+    args: unknown[] = [],
+    gasLimit?: bigint,
+  ): Promise<HexString> {
+    const walletClient = await getWalletClient(this.wagmiConfig, { chainId });
+    if (!walletClient || !walletClient.account?.address) {
+      throw new Error("No wallet client available");
+    }
+
+    try {
+      return await walletClient.writeContract({
+        address: this.contractAddress,
+        abi: this.abi,
+        functionName: methodName,
+        args,
+        gas: gasLimit,
+        account: walletClient.account?.address,
```

Comment:
Why was `account` added? Doesn't this come from the signer of the tx? You added this for something else previously and we were able to remove it. I'm not sure it's needed here either.